docs/programming/testing.rst

```diff
@@ -99,17 +99,20 @@ same versioning system as the test code.
 In some exceptional cases (chicken and egg problems) there is no option to
 generate the data at test time or to keep in the same versioning system as
 the test code.
-For example when you want to run and end to end test to check that you can
+
+For example when you want to run an end to end test to check that you can
 read and write a document from a database, you will have to start with testing
 the read operation against an existing data.
 
-In this case make sure that all pre-generated data is versioned using a well
-defined convention and that the pre-generated data is never modified or
-removed.
-In this way you can still run older versions of the code.
+In this case make sure that all pre-generated data is versioned never modified
+or removed.
+
 Pre-generated data include pre-generated access to the outside resource
-(username / password / server address / port)
+(username / password / server address / port). For usernames make sure to use
```

Comment:
I think that we should not do only "username", but any persistent configuration.

Username, hostnames, folder-names, table-names... etc.

The fact that using a version as a suffic can be on a separate paragraph.

explain that we version this data and each time we need to update a test which runs with changed data, we create a new external resource with -vN+1 and leave the existing one in place